module/VuFind/src/VuFind/Db/Row/User.php

```diff
@@ -232,21 +232,38 @@ protected function encryptOrDecrypt($text, $encrypt = true)
             return $text;
         }
 
+        $configAuth = $this->config->Authentication;
+
         // Load encryption key from configuration if not already present:
-        if (null === $this->encryptionKey) {
-            if (!isset($this->config->Authentication->ils_encryption_key)
-                || empty($this->config->Authentication->ils_encryption_key)
+        if ($this->encryptionKey === null) {
+            if (!isset($configAuth->ils_encryption_key)
+                || empty($configAuth->ils_encryption_key)
             ) {
                 throw new \VuFind\Exception\PasswordSecurity(
                     'ILS password encryption on, but no key set.'
                 );
             }
-            $this->encryptionKey = $this->config->Authentication->ils_encryption_key;
+
+            $this->encryptionKey = $configAuth->ils_encryption_key;
         }
 
         // Perform encryption:
-        $algo = $this->config->Authentication->ils_encryption_algo ?? 'blowfish';
-        $cipher = new BlockCipher(new Openssl(['algorithm' => $algo]));
+        $algo = $configAuth->ils_encryption_algo ?? 'aes-256';
```

Comment:
I wonder if we should define the default encryption algorithm as a constant somewhere (maybe even within this class) so it exists in only one place. Maybe we should also define a key length constant, to help with key generation in a generic way. I think we need to do some work on InstallController::fixSecurityConfiguration, since I don't think it's going to set things up correctly with the new default.

Reply:
Also, upon further reflection, I think we have to leave the default at 'blowfish' here -- otherwise, by upgrading the code, you'll break legacy environments that use blowfish encryption. I think this is the reason we had talked about storing the algorithm in the database... but I think an easier solution is to simply make specifying the algorithm in the config file mandatory going forward. For now, we'll keep the blowfish default. Maybe starting in VuFind 10, we can throw an exception if the algorithm is unspecified in the configuration.